### PR TITLE
hook up various applab import dialogs to redux state

### PR DESCRIPTION
**Pull request by pcardune**
Will see if I can post a many screencast to illustrate how this works.

### CODE REVIEW

File: apps/src/applab/ImportScreensDialog.jsx [hunk 1]
```diff
@@ -249,7 +261,77 @@ var ImportScreensDialog = React.createClass({
...
+  Array.from(dom.children).forEach(child => {
+    if (!elementUtils.isIdAvailable(child.id)) {
+      var existingElement = elementUtils.getPrefixedElementById(child.id);
+      if (existingElement && elementUtils.getId(existingElement.parentNode) !== id) {
+        conflictingIds.push(child.id);
+      }
+    }
+  });
+
+  // TODO: filter out assets that will just be imported without replacing anything.
```
Comment (pcardune) [inline apps/src/applab/ImportScreensDialog.jsx:283]: I could either do this as part of this pull request, or a future pull request. The feature is hidden behind an experiment so nobody will see it in a half-implemented state.

File: apps/src/applab/ImportScreensDialog.jsx [hunk 2]
```diff
@@ -249,7 +261,77 @@ var ImportScreensDialog = React.createClass({
...
+}
+
+export default connect(
+  state => ({
+    isOpen: state.screens.isImportingScreen && state.screens.importProject.fetchedProject,
+    project: getImportableProject(state.screens.importProject.fetchedProject),
+  }),
+  dispatch => ({
+    onImport() {
+      alert("The import button has not been implemented yet. sorry!");
```
Comment (pcardune) [inline apps/src/applab/ImportScreensDialog.jsx:330]: Ditto about this part. Could be a future PR or this PR.

File: apps/src/applab/ImportScreensDialog.jsx [hunk 3]
```diff
@@ -193,26 +197,34 @@ var ImportScreensDialog = React.createClass({
...
       <Dialog
         title={`Import from Project: ${this.props.project.name}`}
-        confirmText="Import"
-        onConfirm={() => this.props.onImport(this.state.selectedScreens,
-                                               this.state.selectedAssets)}
+        confirmText={canImport ? "Import" : null}
+        onConfirm={canImport ? () => this.props.onImport(this.state.selectedScreens,
+                                                         this.state.selectedAssets)
+                                 : null}
+        onCancel={canImport ? null : this.props.handleClose}
```
Comment (islemaster) [inline apps/src/applab/ImportScreensDialog.jsx]: A lot of stuff is `null` if `canImport` is false.  What is the behavior of this dialog if there are no importable screens or assets?  Should it just be rendering a whole different Dialog component in that case?

File: apps/src/applab/ImportScreensDialog.jsx (hunk 1 above)
Comment (islemaster) [inline apps/src/applab/ImportScreensDialog.jsx:283]: Later is fine.

File: apps/src/applab/constants.js [hunk 4]
```diff
@@ -11,4 +11,5 @@ module.exports = {
   ApplabInterfaceMode: utils.makeEnum('CODE', 'DESIGN', 'DATA'),
   DataView: utils.makeEnum('OVERVIEW', 'PROPERTIES', 'TABLE'),
   ANIMATION_LENGTH_MS: 200,
+  IMPORT_SCREEN: "Import screen...",
```
Comment (islemaster) [inline apps/src/applab/constants.js:14]: Is this an i18n string?  Should probably go in applab/en_us.json.

**Comment by islemaster** [comment]
LGTM

File: apps/src/applab/constants.js (hunk 4 above)
Comment (pcardune) [inline apps/src/applab/constants.js:14]: yeah, it is supposed to be, along with the `NEW_SCREEN` constant above.

File: apps/src/applab/ImportScreensDialog.jsx (hunk 3 above)
Comment (pcardune) [inline apps/src/applab/ImportScreensDialog.jsx]: that's a good suggestion. Will do.

**Pull request merged**